app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -38,76 +70,175 @@ class FireDatabase : Database() {
+        val userData = HashMap<String, Any>()
+        userData.put("email", userModel.getEmailAddress())
+        userData.put("firstname", userModel.getFirstname())
+        userData.put("surname", userModel.getSurname())
+        userData.put("dateOfBirth", userModel.getDateOfBirth().toEpochDay())
+        userData.put("distanceGoal", userModel.getDistanceGoal())
+        userData.put("activityTimeGoal", userModel.getActivityTime())
+        userData.put("nbOfPathsGoal", userModel.getNumberOfPathsGoal())
+
+        val future = CompletableFuture<Boolean>()
+        val userId = getUserId()
+
+        database.child(usersProfileFileName).child(userId).updateChildren(userData)
+            .addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(java.lang.Error("Impossible to init the user data on the database.")) }
+        return future
     }
 
-    override fun setPersonalInfo(
-        username: String,
-        firstname: String,
-        surname: String,
-        dateOfBirth: LocalDate
-    ) {
-        val userAdd = HashMap<String, String>()
-        userAdd.put("firstname", firstname)
-        userAdd.put("surname", surname)
-        updateUserData(userAdd, username)
-
-        //for the date
-        val unixDate: Long = dateOfBirth.toEpochDay()
-        val userDateMap = HashMap<String, Long>()
-        userDateMap.put("dateOfBirth", unixDate)
-        updateUserData(userDateMap, username)
+    override fun getUserAccount(userId: String): CompletableFuture<UserModel> {
+        val future = CompletableFuture<UserModel>()
+
+        database.child(usersProfileFileName).child(userId).get().addOnSuccessListener { userData ->
+            if(userData == null) {
+                future.completeExceptionally(java.lang.Error("There is no user account corresponding to this userId."))
+            }else{
+                if(userAuth==null){
+                    future.completeExceptionally(java.lang.Error("There is no user logged on the app."))
+                }else{
+                    val firstname = userData.child("firstname").value
+                    val surname = userData.child("surname").value
+                    val dateOfBirth = userData.child("dateOfBirth").value
+                    val distanceGoal = userData.child("distanceGoal").value
+                    val activityTimeGoal = userData.child("activityTimeGoal").value
+                    val nbOfPathsGoal = userData.child("nbOfPathsGoal").value
+                    if(firstname==null||surname==null||dateOfBirth==null||distanceGoal==null||activityTimeGoal==null||nbOfPathsGoal==null){
+                        future.completeExceptionally(java.lang.Error("The user account present on the database is incomplete."))
+                    }else{
+                        future.complete(
+                            UserModel(userAuth, firstname as String, surname as String, LocalDate.ofEpochDay(dateOfBirth as Long),
```

Comment:
We end up fetching the username two times due to the way the `UserModel` constructor works.